app/models/friendship.rb

```diff
@@ -0,0 +1,45 @@
+class Friendship < ApplicationRecord
+  belongs_to :user
+  belongs_to :friend, class_name: "User"
+  validates :friend_id, uniqueness: { scope: :user_id, message: "Already Sent" }
+  scope :friends, -> {where(status: true)}
+  #scope :friends, -> {where(friend_id: User.current.id).or(where( user_id: User.current.id)).and(where(status: true))}
+
+  def exist_friend_id
+    if !Friendship.exists?(friend_id: id)
+      errors.add(:base, 'No- Request Exists')
+    end
+  end
+  
+  after_update :update_inverse , unless: :has_inverse?
```

Comment:
This can be placed on above of all the methods
The conventional structure of a model is to have 
1. relationships on top (if there any shared concerns/modules that will come top of it)
2. Any global constants
3. scopes
4. validations
5. callbacks
then all the methods below 

Example: 
```ruby
class Bookmark < ApplicationRecord
  include Shareable

  belongs_to :video
  belongs_to :user
  has_one :team, through: :video

  SOME_GLOBAL_CONSTANT = 13

  scope :by_tag, ->(tag) { where(":tag = ANY (tags)", tag: tag) if tag.present? }

  validates :title, presence: true

  after_create :sample_method
  
  def sample_method
    return true
  end
end